### DISCUSSION TITLE
Tooltip changes the width of children components

### DISCUSSION THREAD

**Discussion by victorfunes** [themesberg/flowbite-react#939] [Q&A]
- [X] I have searched the [Issues](https://github.com/themesberg/flowbite-react/issues) to see if this bug has already been reported
- [X] I have tested the latest version

## Steps to reproduce

```
      <div className="bg-gray-300 w-full">
        <Tooltip content="textWidth">
          <button type="button" className="w-full bg-blue-500 text-white px-2">
            Width test
          </button>
        </Tooltip>
      </div>
```
Generates this:
<img width="278" alt="image" src="https://github.com/themesberg/flowbite-react/assets/108006690/751ddf52-ba10-4816-9dbc-9eb1474ee8c3">

Without the tooltip:

```
      <div className="bg-gray-300 w-full">
        <button type="button" className="w-full bg-blue-500 text-white px-2">
          Width test
        </button>
      </div>
```
Generates this:
<img width="278" alt="image" src="https://github.com/themesberg/flowbite-react/assets/108006690/483f98fa-00e2-4286-9be4-cecdcbfb521d">

## Current behavior

Now using the tooltip component changes the width of the children elements, and it shouldn't affect the style.

## Expected behavior

The style of the Tooltip children components should be exactly the same than without the Tooltip.

**Comment by DavideFrancescon** [comment]
This is caused by the [tooltip style](https://www.flowbite-react.com/docs/components/tooltip#theme) and i'm sure is wanted.
Not always you set a div with size as the father of the tooltip, so when this is not present if it hasn't a w-fit, it's going to take up the entire page and cause issues. A solution would be to override the theme or add a className with w-full to override the w-fit.

**Comment by victorfunes** [comment]
Sorry, but I don't understand this behaviour as something wanted. 
If you ask me what would I expect and desire from a Tooltip is that this component not affecting at all the screen and component design (as if it was not present).
Is there not a technical way to avoid the problem you mention without changing the children styles?

In my case, following your comment, I am going to use ```theme={{target: ""}}```by the moment. 
Thank you!

**Comment by tulup-conner** [comment]
I agree with you that this is not the behavior I would expect. I'm sure there is a reason we have `w-fit` as the default here for `tooltip.target`, but I don't know why. It would be helpful if you could try removing `w-fit` from the theme on a fork and see if it breaks any of our examples on the [Tooltip docs](https://www.flowbite-react.com/docs/components/tooltip) unexpectedly. If not, then I can't really see a reason to not just remove the default `w-fit` from the default theme.

**Comment by DavideFrancescon** [comment]
Just tested in [tailwind playground](https://play.tailwindcss.com/BOuObnFuEx). I don't know if it's caused by tailwind or it's default html behavior, but the default behavior of a div is `w-fit`. So i'm pretty sure that just removing that won't change a thing.

**Comment by victorfunes** [comment]
Just if it could be useful for you: I changed in my application theme and everything that was broken after using Tooltips is working now.

```
  tooltip: {
    target: "", // To avoid changing children size as default theme is using w-fit
  },
```

**Comment by DavideFrancescon** [comment]
I guess i'll have to test everything then. I'll do it tomorrow and update you all on the results.

**Comment by DavideFrancescon** [comment]
<img width="1308" alt="Screenshot 2023-08-26 alle 10 46 15" src="https://github.com/themesberg/flowbite-react/assets/102860588/efa09562-2a92-4267-9dbb-d365429b3225">
@victorfunes @tulup-conner 
Removing the w-fit breaks everything. It might work in some cases, but not in all of them, so I'm sure that removing it will surely break some site using the default settings.

**Comment by victorfunes** [comment]
Thank you for your quick action and answer @DavideFrancescon

**Comment by tulup-conner** [comment]
Thanks. I'll convert this into a discussion for the future, then. This is probably a quirk with Floating UI, which we might be able to fix by moving `w-fit` further down the tree.

**Comment by MiroslavPetrik** [comment]
It affects me as well. Definitely the `width: fit-content` css rule. I need my children wrapped by `Tooltip` to be full-width as well.

**Comment by nardocesar** [comment]
I ran into the same issue and the workaround I found is customizing `theme` prop like this:

```jsx
<Tooltip content="Tooltip content" theme={{ target: '' }}>
```

**Comment by Schabaani** [comment]
In 2026, same issue! `theme={{ target: '' }}` didn't work for me

```
export const Guard = ({
  disabled,
  tooltip,
  children,
}: PropsWithChildren<TProps>) => {
  const t = useTranslations();
  if (disabled) {
    return (
      <Tooltip content={tooltip ?? t('notAllowed')} theme={{target: ''}}>
        {children}
      </Tooltip>
    );
  }
  return children;
};
```

<img width="223" height="115" alt="Screenshot 2026-02-12 at 11 31 11" src="https://github.com/user-attachments/assets/18359304-90ea-4a12-b378-c2462c1b292f" />
